components/use.math.ts

```diff
@@ -9,28 +9,14 @@ export const useMath = (): ((
 ) => boolean) => {
   return (expression, values) => {
     const parser = init(formula, (term: string) => {
-      if (values[term]) {
+      if (values.hasOwnProperty(term)) {
         return values[term]
       }
 
       throw new Error(`Invalid term: ${term}`);
     })
 
     try {
-      let processed = expression
-
-      Object.keys(values).forEach((key) => {
```

Comment:
I think this block is still needed to replace the values within the expression -> so that values in the condition like $FIELD1 will get replaced correctly with the value of the current form submission

Reply:
I dont think so. The processed variable is never used. It would be the version where $FIELD1 is rpelaced with the value.

Instead the ini function of the parser (Line 11) is the one that does the mapping. 

Whenever parser.expressionToValue happens on an expression it does not understand itself, the formular function on line 11-14 is called. This will then map all variables of ohmyform to the repsective value. If OhMyForm also does not know the expression the "Invalid Term X" Expression is thrown.

Reply:
good point